### ISSUE TITLE
autoreload hang for changed instance methods: between 7.5.0 and 7.8.0

### ISSUE THREAD

**Issue by histed** [ipython/ipython#11885]
Possible autoreload regression in ipython 7.8.0 and jupyterlab 1.0+

With %autoreload 2, when I change a method in a source file when holding an existing instance object in memory, autoreload used to reload the changed instance code just fine.  (It didn't find new methods, which required an explicit imp.reload(), but that's ok.)

With ipython 7.8.0, changing code in a method causes the kernel to hang when autoreload triggered by cell execute.  Interrupting kernel gives following traceback

````
[autoreload of testcase failed: Traceback (most recent call last):
  File "/Users/histed/anaconda37/envs/mh37b/lib/python3.7/site-packages/IPython/extensions/autoreload.py", line 245, in check
    superreload(m, reload, self.old_objects)
  File "/Users/histed/anaconda37/envs/mh37b/lib/python3.7/site-packages/IPython/extensions/autoreload.py", line 450, in superreload
    update_generic(old_obj, new_obj)
  File "/Users/histed/anaconda37/envs/mh37b/lib/python3.7/site-packages/IPython/extensions/autoreload.py", line 387, in update_generic
    update(a, b)
  File "/Users/histed/anaconda37/envs/mh37b/lib/python3.7/site-packages/IPython/extensions/autoreload.py", line 357, in update_class
    update_instances(old, new)
  File "/Users/histed/anaconda37/envs/mh37b/lib/python3.7/site-packages/IPython/extensions/autoreload.py", line 312, in update_instances
    update_instances(old, new, obj.__dict__, visited)
  File "/Users/histed/anaconda37/envs/mh37b/lib/python3.7/site-packages/IPython/extensions/autoreload.py", line 317, in update_instances
    update_instances(old, new, obj, visited)
  File "/Users/histed/anaconda37/envs/mh37b/lib/python3.7/site-packages/IPython/extensions/autoreload.py", line 317, in update_instances
    update_instances(old, new, obj, visited)
  File "/Users/histed/anaconda37/envs/mh37b/lib/python3.7/site-packages/IPython/extensions/autoreload.py", line 317, in update_instances
    update_instances(old, new, obj, visited)
  File "/Users/histed/anaconda37/envs/mh37b/lib/python3.7/site-packages/IPython/extensions/autoreload.py", line 315, in update_instances
    if hasattr(obj, 'items') or (hasattr(obj, '__contains__')
KeyboardInterrupt
]
````

testcase.py:

````python
class Class1:
    def __init__(self):
        print(0)

    def method(self):
        print(1)
        #print(2)
````

In a notebook, run:
````python
import testcase

c = testcase.Class1()
c.method()
````

Then change method() by commenting/uncommenting print(2).  Then run

````
c.method()
````

And kernel hangs.

**Comment by histed** [comment]
may be related to https://github.com/ipython/ipython/pull/11644  @daharn @Carreau worked on

**Comment by daharn** [comment]
Should be solved in 7.9 or if you install the latest version.

**Comment by histed** [comment]
confirm: latest master on github fixes this.
